### PR TITLE
Updated .format() to f-strings

### PR DESCRIPTION
**Pull request by dlalap**
- [ ] references #29547
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/html.py [hunk 1]
```diff
@@ -617,7 +617,7 @@ def _build_xpath_expr(attrs) -> str:
         attrs["class"] = attrs.pop("class_")
 
     s = [f"@{k}={repr(v)}" for k, v in attrs.items()]
-    return "[{expr}]".format(expr=" and ".join(s))
+    return f"[{' and '.join(s)}]"
```
Comment (jbrockmendel) [inline pandas/io/html.py]: can you move the join call to the previous line

File: pandas/io/html.py [hunk 2]
```diff
@@ -846,7 +846,7 @@ def _parser_dispatch(flavor):
 
 
 def _print_as_set(s) -> str:
-    return "{" + "{arg}".format(arg=", ".join(pprint_thing(el) for el in s)) + "}"
+    return f"{{{', '.join(pprint_thing(el) for el in s)}}}"
```
Comment (jbrockmendel) [inline pandas/io/html.py]: join on previous line pls

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good if you can address @jbrockmendel suggestions and I think CI failures are unrelated / have been fixed on master since this was created.

To be sure, can you locally on your `f-strings` branch run:

```sh
git fetch upstream
git merge upstream/master
git push origin f-strings
```

I think should get CI green for this. Flag me down at PyData if you have any questions

**Review by WillAyd — APPROVED** [review]
lgtm @jbrockmendel

File: pandas/io/html.py [hunk 3]
```diff
@@ -846,7 +846,8 @@ def _parser_dispatch(flavor):
 
 
 def _print_as_set(s) -> str:
-    return "{" + "{arg}".format(arg=", ".join(pprint_thing(el) for el in s)) + "}"
+    arg = ", ".join(pprint_thing(el) for el in s)
+    return f"{{{arg}}}"
```
Comment (jbrockmendel) [inline pandas/io/html.py]: is this adding a layer of braces?

Comment (jbrockmendel) [reply pandas/io/html.py]: you might be able to get rid of the pprint_thing, not sure

File: pandas/io/html.py [hunk 4]
```diff
@@ -616,8 +616,8 @@ def _build_xpath_expr(attrs) -> str:
     if "class_" in attrs:
         attrs["class"] = attrs.pop("class_")
 
-    s = [f"@{k}={repr(v)}" for k, v in attrs.items()]
-    return "[{expr}]".format(expr=" and ".join(s))
+    s = " and ".join([f"@{k}={repr(v)}" for k, v in attrs.items()])
```
Comment (jbrockmendel) [inline pandas/io/html.py]: if you end up making another pass through this file, can you do this in two lines?  i.e. keep `s` as it is now and do the join in a separate step

File: pandas/io/html.py (hunk 3 above)
Comment (WillAyd) [reply pandas/io/html.py]: The double brace is how to add a literal brace in an f-string, so looks OK

**Comment by WillAyd** [comment]
Thanks @dlalap congrats on first PR. If you'd like to address some suggestions from @jbrockmendel in a follow up would certainly welcome, along with any other contributions

**Pull request merged**